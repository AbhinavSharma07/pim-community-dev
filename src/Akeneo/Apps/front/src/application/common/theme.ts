import {ThemedStyledProps} from 'styled-components';

export const theme = {
    color: {
        blue: '#5992c7',
        darkBlue: '#11324d',
        grey: '#a1a9b7',
        slateGrey: '#67768a',
    },
};

export type PropsWithTheme<P = {}> = ThemedStyledProps<P, typeof theme>;
