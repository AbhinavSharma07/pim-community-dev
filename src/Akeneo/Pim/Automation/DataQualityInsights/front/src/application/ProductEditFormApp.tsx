import React, {FunctionComponent} from 'react';
import {Provider} from 'react-redux';
import {productEditFormStore} from '../infrastructure/store';
import {CatalogContextListener, PageContextListener, ProductContextListener} from './listener';
import {Product} from '../domain';
import {fetchProduct, fetchProductDataQualityEvaluation} from '../infrastructure/fetcher';
import {AxisRatesOverviewPortal} from './component/ProductEditForm';
import {AxesContextProvider} from './context/AxesContext';
import {DataQualityInsightsTabContent} from './component/ProductEditForm/TabContent';
import AttributesTabContent from './component/ProductEditForm/TabContent/AttributesTabContent';
import {pimTheme} from 'akeneo-design-system';
import {ThemeProvider} from 'styled-components';
import AxisEvaluation from './component/ProductEditForm/TabContent/DataQualityInsights/AxisEvaluation';
import Criterion from "./component/ProductEditForm/TabContent/DataQualityInsights/Criterion";

interface ProductEditFormAppProps {
  catalogChannel: string;
  catalogLocale: string;
  product: Product;
}

const ProductEditFormApp: FunctionComponent<ProductEditFormAppProps> = ({product, catalogChannel, catalogLocale}) => {
  return (
    <ThemeProvider theme={pimTheme}>
      <Provider store={productEditFormStore}>
        <CatalogContextListener catalogChannel={catalogChannel} catalogLocale={catalogLocale} />
        <PageContextListener />
        <ProductContextListener product={product} productFetcher={fetchProduct}/>

        <AttributesTabContent product={product}/>

        <AxesContextProvider axes={['enrichment']}>
          <DataQualityInsightsTabContent product={product} productEvaluationFetcher={fetchProductDataQualityEvaluation}>
            <AxisEvaluation axis={'enrichment'}>
              <Criterion code={'completeness_of_required_attributes'} />
              <Criterion code={'completeness_of_non_required_attributes'} />
              <Criterion code={'missing_image_attribute'} />
            </AxisEvaluation>
          </DataQualityInsightsTabContent>
          <AxisRatesOverviewPortal />
        </AxesContextProvider>
      </Provider>
    </ThemeProvider>
  );
};

export default ProductEditFormApp;
