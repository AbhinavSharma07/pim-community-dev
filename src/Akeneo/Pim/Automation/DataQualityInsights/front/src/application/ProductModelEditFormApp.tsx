import React, {FunctionComponent} from 'react';
import {Provider} from 'react-redux';
import {productEditFormStore} from '../infrastructure/store';
import {CatalogContextListener, PageContextListener, ProductContextListener} from './listener';
import {Product} from '../domain';
import fetchProductModel from '../infrastructure/fetcher/ProductEditForm/fetchProductModel';
import {DataQualityInsightsTabContent} from './component/ProductEditForm/TabContent';
import fetchProductModelEvaluation from '../infrastructure/fetcher/ProductEditForm/fetchProductModelEvaluation';
import {AxesContextProvider} from './context/AxesContext';
import AttributesTabContent from './component/ProductEditForm/TabContent/AttributesTabContent';
import {AssetCollectionIcon, EditIcon, pimTheme} from 'akeneo-design-system';
import {ThemeProvider} from 'styled-components';
import AxisEvaluation from './component/ProductEditForm/TabContent/DataQualityInsights/AxisEvaluation';
import {Criterion, Icon} from './component/ProductEditForm/TabContent/DataQualityInsights/Criterion';
import {Recommendation} from './component/ProductEditForm/TabContent/DataQualityInsights/Recommendation';
import {CRITERION_DONE, CRITERION_NOT_APPLICABLE} from '../domain/Evaluation.interface';
import {isSuccess} from './helper';
import {CRITERION_DONE, CRITERION_NOT_APPLICABLE} from "../domain/Evaluation.interface";
import {isSuccess} from "./helper";

const translate = require('oro/translator');

interface ProductModelEditFormAppProps {
  catalogChannel: string;
  catalogLocale: string;
  product: Product;
}

const ProductModelEditFormApp: FunctionComponent<ProductModelEditFormAppProps> = ({
  product,
  catalogChannel,
  catalogLocale,
}) => {
  return (
    <ThemeProvider theme={pimTheme}>
      <Provider store={productEditFormStore}>
        <CatalogContextListener catalogChannel={catalogChannel} catalogLocale={catalogLocale} />
        <PageContextListener />
        <ProductContextListener product={product} productFetcher={fetchProductModel} />

        <AttributesTabContent product={product} />

        <AxesContextProvider axes={['enrichment']}>
          <DataQualityInsightsTabContent product={product} productEvaluationFetcher={fetchProductModelEvaluation} >
            <AxisEvaluation axis={'enrichment'}>
              <Criterion code={'completeness_of_required_attributes'}>
                <Icon type={EditIcon}/>
              </Criterion>
              <Criterion code={'completeness_of_non_required_attributes'}>
                <Icon type={EditIcon}/>
              </Criterion>
              <Criterion code={'enrichment_image'}>
                <Icon type={AssetCollectionIcon}/>
                <Recommendation supports={criterion => criterion.status === CRITERION_NOT_APPLICABLE || (criterion.status === CRITERION_DONE && !isSuccess(criterion.rate) && criterion.improvable_attributes.length === 0)}>
                  <span className="NotApplicableAttribute">{translate('akeneo_data_quality_insights.product_evaluation.messages.add_image_attribute_recommendation')}</span>
                </Recommendation>
              </Criterion>
            </AxisEvaluation>
          </DataQualityInsightsTabContent>
        </AxesContextProvider>
      </Provider>
    </ThemeProvider>
  );
};

export default ProductModelEditFormApp;
