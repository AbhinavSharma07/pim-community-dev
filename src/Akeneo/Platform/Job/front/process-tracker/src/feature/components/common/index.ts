export * from './JobExecutionStatus';
export * from './StopJobAction';
