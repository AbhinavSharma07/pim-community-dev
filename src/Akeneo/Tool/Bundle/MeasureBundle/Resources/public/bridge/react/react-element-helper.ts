import ReactDOM from 'react-dom';

let container: Element | null = null;

const mountReactElementRef = (component: JSX.Element) => {
  if (null === container) {
    container = document.createElement('div');
    // (container as any).style.height = '100%';
    // (container as any).style.display = 'flex';
    // (container as any).style.flexDirection = 'column';
    ReactDOM.render(component, container);
  }

  return container;
};

const unmountReactElementRef = () => {
  if (null !== container) {
    ReactDOM.unmountComponentAtNode(container);
    container = null;
  }
};

export {mountReactElementRef, unmountReactElementRef};
