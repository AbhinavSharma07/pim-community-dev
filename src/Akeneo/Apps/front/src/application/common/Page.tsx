import * as React from 'react';

import {PimView} from '../../infrastructure/PimView';

type Props = React.PropsWithChildren<{
    breadcrumb?: any;
    buttons?: React.ReactElement[];
    title?: any;
}>;

export const Page = ({children, breadcrumb, buttons, title}: Props) => (
    <div className='AknDefault-contentWithColumn'>
        <div className='AknDefault-thirdColumnContainer'>
            <div className='AknDefault-thirdColumn' />
        </div>

        <div className='AknDefault-contentWithBottom'>
            <div className='AknDefault-mainContent'>
                <header className='AknTitleContainer'>
                    <div className='AknTitleContainer-line'>
                        <div className='AknTitleContainer-mainContainer'>
                            <div className='AknTitleContainer-line'>
                                <div className='AknTitleContainer-breadcrumbs'>{breadcrumb}</div>
                                <div className='AknTitleContainer-buttonsContainer'>
                                    <PimView
                                        className='AknTitleContainer-userMenuContainer AknTitleContainer-userMenu'
                                        viewName='pim-apps-user-navigation'
                                    />
                                    {buttons && (
                                        <div className='AknTitleContainer-actionsContainer AknButtonList'>
                                            {buttons.map((button, index) => (
                                                <React.Fragment key={index}>{button}</React.Fragment>
                                            ))}
                                        </div>
                                    )}
                                </div>
                            </div>

                            <div className='AknTitleContainer-line'>
                                <div className='AknTitleContainer-title'>{title}</div>
                                <div className='AknTitleContainer-state' />
                            </div>

                            <div className='AknTitleContainer-line'>
                                <div className='AknTitleContainer-context AknButtonList' />
                            </div>

                            <div className='AknTitleContainer-line'>
                                <div className='AknTitleContainer-meta AknButtonList' />
                            </div>
                        </div>
                    </div>

                    <div className='AknTitleContainer-line'>
                        <div className='AknTitleContainer-navigation' />
                    </div>

                    <div className='AknTitleContainer-line'>
                        <div className='AknTitleContainer-search' />
                    </div>
                </header>

                {children}
            </div>
        </div>
    </div>
);
