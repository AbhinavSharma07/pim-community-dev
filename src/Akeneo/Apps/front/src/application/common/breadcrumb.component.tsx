import * as React from 'react';
import {withRouter} from '../shared/router/router.hoc-factory';
import {Router} from '../shared/router/router.interface';

class InvalidItemTypeError extends Error {}

interface BreadcrumbItem {
    action: {
        type: string;
        route?: string;
        parameters?: {[key: string]: string | number};
    };
    label: string;
}

const renderItem = (router: Router, item: BreadcrumbItem, key: number, last: boolean) => {
    switch (item.action.type) {
        case 'redirect':
            return renderRedirect(router, item, key, last);
        case 'display':
            return renderDisplay(item, key, last);
        default:
            throw new InvalidItemTypeError(
                `The action type "${item.action.type}" is not supported by the Breadcrumb component`
            );
    }
};

const renderRedirect = (router: Router, item: BreadcrumbItem, key: number, last: boolean) => {
    const path = `#${router.generate(
        item.action.route as string,
        item.action.parameters ? item.action.parameters : {}
    )}`;

    return (
        <a
            key={key}
            onClick={(event: React.MouseEvent<HTMLAnchorElement>) => {
                event.preventDefault();

                if (path !== window.location.hash) {
                    router.redirect(path, {trigger: true});
                }

                return false;
            }}
            title={item.label}
            href={path}
            className={`AknBreadcrumb-item AknBreadcrumb-item--routable ${last ? 'AknBreadcrumb-item--final' : ''}`}
        >
            {item.label}
        </a>
    );
};

const renderDisplay = (item: BreadcrumbItem, key: number, last: boolean) => {
    return (
        <span key={key} className={`AknBreadcrumb-item ${last ? 'AknBreadcrumb-item--final' : ''}`} title={item.label}>
            {item.label}
        </span>
    );
};

export const Breadcrumb = withRouter(({router, items}: {router: Router; items: BreadcrumbItem[]}) => {
    return (
        <div className='AknBreadcrumb'>
            {items.map((item: BreadcrumbItem, key: number) => renderItem(router, item, key, key === items.length - 1))}
        </div>
    );
});
