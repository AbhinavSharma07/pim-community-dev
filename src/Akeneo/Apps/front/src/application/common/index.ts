import {Breadcrumb} from './breadcrumb/Breadcrumb';
import {Item as BreadcrumbItem} from './breadcrumb/Item';
import {ApplyButton} from './button/ApplyButton';
import {Header} from './Header';
import {Page} from './Page';

export {Header, Page, Breadcrumb, BreadcrumbItem, ApplyButton};
