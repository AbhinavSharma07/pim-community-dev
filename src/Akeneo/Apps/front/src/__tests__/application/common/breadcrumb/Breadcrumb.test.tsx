import * as React from 'react';
import {create} from 'react-test-renderer';
import {Breadcrumb, BreadcrumbItem} from '../../../../application/common';

describe('Button component', () => {
    test('Matches the snapshot', () => {
        const component = create(
            <Breadcrumb>
                <BreadcrumbItem label='Is a link' onClick={() => undefined} />
                <BreadcrumbItem label='Not a link' />
                <BreadcrumbItem label='Is a link and last item' onClick={() => undefined} />
            </Breadcrumb>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
});
