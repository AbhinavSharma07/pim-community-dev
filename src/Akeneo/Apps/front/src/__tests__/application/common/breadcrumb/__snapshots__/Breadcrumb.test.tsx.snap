// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<div
  className="AknBreadcrumb"
>
  <span
    className="AknBreadcrumb-item AknBreadcrumb-item--routable"
    onClick={[Function]}
    title="Is a link"
  >
    Is a link
  </span>
  <span
    className="AknBreadcrumb-item"
    title="Not a link"
  >
    Not a link
  </span>
  <span
    className="AknBreadcrumb-item AknBreadcrumb-item--final AknBreadcrumb-item--routable"
    onClick={[Function]}
    title="Is a link and last item"
  >
    Is a link and last item
  </span>
</div>
`;
